src/server/pps/server/s3g_sidecar_test.go

```diff
@@ -667,7 +665,7 @@ func TestS3SkippedDatums(t *testing.T) {
 		pipelineCommit := client.NewCommit(pipeline, "master", "")
 		// Add files to 'repo'. Old files in 'repo' should be reprocessed in every
 		// job, changing the 'background' field in the output
-		for i := 0; i < 10; i++ {
+		for i := 0; i < 5; i++ {
```

Comment:
this just takes a long time. Reducing the number of iterations seem to make it more manageable

Reply:
This test is probably exposing a real performance issue we need to address. I think we should keep this as skipped with the original test. I am wrapping up a few performance fixes that may address this. One of the more significant ones is in master, so maybe you should merge that into this PR and see if that helps.